spynnaker/pyNN/models/neuron/population_machine_vertex.py

```diff
@@ -384,6 +397,227 @@ def get_binary_file_name(self):
     def get_binary_start_type(self):
         return ExecutableType.USES_SIMULATION_INTERFACE
 
+    @inject_items({
+        "machine_time_step": "MachineTimeStep",
+        "time_scale_factor": "TimeScaleFactor",
+        "application_graph": "MemoryApplicationGraph",
+        "machine_graph": "MemoryMachineGraph",
+        "routing_info": "MemoryRoutingInfos",
+        "data_n_time_steps": "DataNTimeSteps",
+        "n_key_map": "MemoryMachinePartitionNKeysMap"
+    })
+    @overrides(
+        AbstractGeneratesDataSpecification.generate_data_specification,
+        additional_arguments={
+            "machine_time_step", "time_scale_factor",
+            "application_graph", "machine_graph", "routing_info",
+            "data_n_time_steps", "n_key_map"
+        })
+    def generate_data_specification(
+            self, spec, placement, machine_time_step, time_scale_factor,
+            application_graph, machine_graph, routing_info, data_n_time_steps,
+            n_key_map):
+        """
+        :param machine_time_step: (injected)
+        :param time_scale_factor: (injected)
+        :param application_graph: (injected)
+        :param machine_graph: (injected)
+        :param routing_info: (injected)
+        :param data_n_time_steps: (injected)
+        :param n_key_map: (injected)
+        """
+        # pylint: disable=too-many-arguments, arguments-differ
+
+        spec.comment("\n*** Spec for block of {} neurons ***\n".format(
```

Comment:
comment needs to be for a neuron and not neurons!

Reply:
thats the same as it was from the app point. and its still covering a machine vertex worth of neuronS